analytics_data_api/management/commands/generate_data.py

```diff
@@ -0,0 +1,349 @@
+                      models.CourseProgramMetadata]:
+            model.objects.using(database).all().delete()
+
+        logger.info("Deleted all daily course enrollment data.")
+        logger.info("Generating new daily course enrollment data...")
+
+    logger.info("Generating new daily course enrollment data...")
+
+    # Create new data
+    daily_total = 1500
+    date = start_date
+
+    enrollment_objs = models.CourseEnrollmentModeDaily.objects.using(database).filter(course_id=course_id)
+    # start incrementing cumulative count, as opposed to starting from 0 every time we run the command
+    cumulative_count = enrollment_objs.latest().cumulative_count \
+        if (use_current_cumulative and enrollment_objs) \
+        else 0
+
+    progress = tqdm(total=(end_date - date).days + 2)
+    while date <= end_date:
+        _generate_day(date, course_id, daily_total, cumulative_count, database, add_birth_year)
+        progress.update(1)
+        date = date + datetime.timedelta(days=1)
+
+    models.CourseMetaSummaryEnrollment.objects.using(database).filter(course_id=course_id).delete()
+    for index, (mode, ratio) in enumerate(ENROLLMENT_MODE_RATIOS.items()):
+        count = int(ratio * daily_total)
+        pass_rate = min(random.normalvariate(.45 + (.1 * index), .15), 1.0)
+        cumulative_count = count + random.randint(0, 100)
+        models.CourseMetaSummaryEnrollment.objects.using(database).create(
+            course_id=course_id, catalog_course_title='Demo Course', catalog_course='Demo_Course',
+            start_time=timezone.now() - datetime.timedelta(weeks=6),
+            end_time=timezone.now() + datetime.timedelta(weeks=10),
+            pacing_type='self_paced', availability='Starting Soon', enrollment_mode=mode, count=count,
+            cumulative_count=cumulative_count, count_change_7_days=random.randint(-50, 50),
+            passing_users=int(cumulative_count * pass_rate))
+
+    progress.update(1)
+    progress.close()
+    logger.info("Done generating daily enrollment data")
+
+
+def generate_weekly_data(course_id, start_date, end_date, database, delete_data=True):
+    activity_types = ['PLAYED_VIDEO', 'ATTEMPTED_PROBLEM', 'POSTED_FORUM']
+
+    # Ensure we start on a Wednesday 00:00
+    days_ahead = -start_date.weekday() + 2
+    start = (start_date + datetime.timedelta(days_ahead)).replace(hour=0, minute=0, second=0, microsecond=0)
+
+    if delete_data:
+        models.CourseActivityWeekly.objects.using(database).all().delete()
+        logger.info("Deleted all weekly course activity.")
+
+    logger.info("Generating new weekly course activity data...")
+
+    progress = tqdm(total=math.ceil((end_date - start).days / 7.0))
+    while start < end_date:
+        active_students = random.randint(100, 4000)
+        # End date should occur on Wednesday at 00:00:00
+        end = (start + datetime.timedelta(weeks=1)).replace(hour=0, minute=0, second=0, microsecond=0)
```

Comment:
shouldn't need to do the replace here, midnight + 1 week is still midnight

ugh except for daylight savings, how does that work with python timedelta addition
although that still doesn't work here if you lose an hour and end up on tuesday and then reset to start of tuesday
unless the datetime is UTC? which doesn't have daylight savings. In that case it's ok.

really you want to be working in dates not microsecond precision timedelta, so you can say wednesday + 7 days like a human would. But that's something that has to be done across the codebase.

Reply:
Yeah, that makes sense. Updated to just add the seven days.